xbmc/video/windows/GUIWindowVideoBase.cpp

```diff
@@ -1080,20 +1080,21 @@ bool CGUIWindowVideoBase::OnPlayMedia(int iItem, const std::string &player)
   CServiceBroker::GetPlaylistPlayer().Reset();
   CServiceBroker::GetPlaylistPlayer().SetCurrentPlaylist(PLAYLIST::TYPE_NONE);
 
-  CFileItem item(*pItem);
+  CFileItemPtr pItemCopy = std::make_shared<CFileItem>(*pItem);
+
   if (pItem->IsVideoDb())
   {
-    item.SetPath(pItem->GetVideoInfoTag()->m_strFileNameAndPath);
-    item.SetProperty("original_listitem_url", pItem->GetPath());
+    pItemCopy->SetPath(pItem->GetVideoInfoTag()->m_strFileNameAndPath);
+    pItemCopy->SetProperty("original_listitem_url", pItem->GetPath());
   }
-  CLog::Log(LOGDEBUG, "{} {}", __FUNCTION__, CURL::GetRedacted(item.GetPath()));
+  CLog::Log(LOGDEBUG, "{} {}", __FUNCTION__, CURL::GetRedacted(pItemCopy->GetPath()));
 
-  item.SetProperty("playlist_type_hint", m_guiState->GetPlaylist());
+  pItemCopy->SetProperty("playlist_type_hint", m_guiState->GetPlaylist());
 
   if (m_thumbLoader.IsLoading())
     m_thumbLoader.StopAsync();
 
-  CServiceBroker::GetPlaylistPlayer().Play(pItem, player);
+  CServiceBroker::GetPlaylistPlayer().Play(pItemCopy, player);
```

Comment:
Although your fix is correct, I think the fix can look a bit simpler. The only line that needs to be changed to fix the issue is this one. Replace ‚pItem‘ with ‚std::make_shared<CFileItem>(&item)‘.

Reply:
On second thought, your approach is better than what I suggested, because my suggestion would introduce a second copy.